src/clang-c-frontend/clang_c_main.cpp

```diff
@@ -67,6 +67,13 @@ bool clang_c_maint::clang_main()
 
   forall_symbol_base_map(it, context.symbol_base_map, main)
   {
+    // if user provided class/contract name
+    if(!config.cname.empty() && !config.main.empty() && main != "main")
```

Comment:
Couldn't there be a class/namespace/etc containing a function called "main"?

Reply:
My mistake. Will fix.